Add app settings example with reducer

Intellemerc opened the pull request:
Also moved order's view's actions to the reducer(this is a more common pattern, to keep reducer and actions together). Another option is to move actions to it's own file

Moved order view state into "app" view state easier to manage in central location. I could also see argument for viewstate in orderview folder but then importing it into central

Intellemerc commented:
@mlo

mlo commented:
It is nice now that there a few use cases. 

1) I don't see the Action change we talked about yet. 
2) For actions in the reducer, I am OK with it; actually, I considered creating a "state" folder instead of "reducers" and "sagas" and thought about putting actions, reducers, sagas, and the state interface into the same file called "ordersViewState.ts". What do you think? The main benefit was selling it as an easy to find where things are. I often hear complaints about "I have to open 10 files to make a small change". Whats your oppinion on this?
3) The only other thing I want to put into this example is a HOC that manages the ordersView and maybe it would show the which break is hit on above to simulate the header. Do you have an idea how this should work?
4) File name  casing. Always camel case? Export casing: Classes upper case, functions camel. 
5) TS interfaces... stick with I prefix?There is some debate and I think TS even gives a warning about, but I see it happening a lot and in thrird party libs and don't mind it.

Thanks for all the changes, I did not even think about the bind pattern (or issue with lamda) and the refactor to use mapStateToProps, etc makes it easier to read.

Intellemerc commented:
1. updated
2. "thought about putting actions, reducers, sagas, and the state interface into the same file called "ordersViewState.ts" - not sure, i feel like that is putting a lot in one file, and it has a lot of different concerns
3. lets discuss tomorrow, I am not picturing what you want?
4. fixed, I was just being lazy
5. I was just following what you had in project, I updated TSLint to force I prefix(my preference)

Intellemerc commented on src/components/Header.tsx:
```diff
@@ -0,0 +1,37 @@
+    cycleResponsive: () => any;
+}
+
+const Header = ({currentResponsive, cycleResponsive}: IProps) =>(
```
Notice I used a functional component here. Generally best practice is unless you need lifecycle methods(componentDidMount, etc) or the React state object functional components are preferred
